table/server/underdb/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java

```diff
@@ -146,70 +155,142 @@ public String getName() {
-      UdbBypassSpec bypassSpec)
-      throws IOException {
-    String tableName = table.getTableName();
-    AlluxioURI ufsUri;
-    AlluxioURI alluxioUri = mUdbContext.getTableLocation(tableName);
-    String hiveUfsUri = table.getSd().getLocation();
+  /**
+   * Mounts the database to Alluxio filesystem. This method mounts tables and partitions
+   * (if applicable) within this database to Alluxio filesystem.
+   *
+   * @param bypassSpec bypass spec
+   */
+  private void mount(UdbBypassSpec bypassSpec) throws IOException {
+    List<String> tableNames = getTableNames();
+    Map<Table, List<Partition>> tables = new HashMap<>(tableNames.size());
+    for (String tableName : tableNames) {
+      Table table;
+      List<Partition> partitions;
+      try (CloseableResource<IMetaStoreClient> client = mClientPool.acquireClientResource()) {
+        table = client.get().getTable(mHiveDbName, tableName);
+        partitions = client.get().listPartitions(mHiveDbName, table.getTableName(), (short) -1);
+        tables.put(table, partitions);
+      } catch (TException e) {
+        throw new IOException(String.format(
+            "Failed to get table %s and partitions of database %s", tableName, mHiveDbName));
+      }
+    }
 
-    try {
-      PathTranslator pathTranslator = new PathTranslator();
-      if (bypassSpec.hasFullTable(tableName)) {
-        pathTranslator.addMapping(hiveUfsUri, hiveUfsUri);
-        return pathTranslator;
+    mount(new ArrayList<>(tables.keySet()), bypassSpec);
+    for (Map.Entry<Table, List<Partition>> entry : tables.entrySet()) {
+      Table table = entry.getKey();
+      List<Partition> partitions = entry.getValue();
+      mount(table, partitions, bypassSpec);
+    }
+  }
+
+  private void mount(List<Table> tables, UdbBypassSpec bypassSpec)
+      throws IOException {
+    HashMultimap<AlluxioURI, Table> colocatedTables = HashMultimap.create();
+    for (Table table : tables) {
+      if (table.getSd() == null || table.getSd().getLocation() == null) {
+        continue;
+      }
+      String tableUfsPath = table.getSd().getLocation();
+      if (bypassSpec.hasFullTable(table.getTableName())) {
+        mPathTranslator.addMapping(tableUfsPath, tableUfsPath);
+        continue;
       }
-      ufsUri = new AlluxioURI(table.getSd().getLocation());
-      pathTranslator.addMapping(
-          UdbUtils.mountAlluxioPath(tableName,
-              ufsUri,
-              alluxioUri,
-              mUdbContext,
-              mConfiguration),
-          hiveUfsUri);
+      AlluxioURI tableUfsUri = new AlluxioURI(tableUfsPath);
+      AlluxioURI rootUfsUri = new AlluxioURI(tableUfsUri, "/", false);
+      colocatedTables.put(rootUfsUri, table);
+    }
+    for (Map.Entry<AlluxioURI, Collection<Table>> entry : colocatedTables.asMap().entrySet()) {
+      AlluxioURI rootUfsUri = entry.getKey();
+      Collection<Table> childTables = entry.getValue();
```

Comment:
It seems you defined this variable but never used it

Reply:
Removed in 047139a5.